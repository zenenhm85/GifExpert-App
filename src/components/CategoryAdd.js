import React, { useState } from "react";
import PropTypes from "prop-types";

export const CategoryAdd = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((categories) => {
        const find = categories.find((item) => item === inputValue);        
        if (!find) {
          setInputValue("");
          return [inputValue, ...categories];
          // setCategories([...categories,`Hunter ${categories.length}`]); This way is also good
        } else {            
            return categories;
        }
      });
      
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        className="input-categories"
        value={inputValue}
        onChange={handleInputChange}
        placeholder="Insert one category"
      />
    </form>
  );
};
CategoryAdd.propTypes = {
    setCategories: PropTypes.func.isRequired,
};