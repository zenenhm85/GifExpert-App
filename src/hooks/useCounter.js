import { useState } from "react";

export const useCounter = (initialState = 10) => {
  const [counter, setCounter] = useState(initialState);

  const increment = (factor) => {
    setCounter(counter + factor);
  };
  const decrement = (factor) => {
    setCounter(counter - factor);
  };
  const reset = () => {
    setCounter(initialState);
  };

  return {
    counter,
    increment,
    decrement,
    reset,
  };
};
