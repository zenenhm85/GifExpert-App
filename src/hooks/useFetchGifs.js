import { useState, useEffect } from "react";
import { getGifts } from "../helpers/getGifts";

export const useFetchGifs = (category) => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getGifts(category).then((imgs) => {
      setState({
        data: imgs,
        loading: false,
      });
    });
  }, [category]);

  return state; // { data:[], loading: true };
};
