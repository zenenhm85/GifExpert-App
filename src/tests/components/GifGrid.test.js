import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Tests for GifGrid", () => {
  const category = "One Punch";

  test("should display the component correctly", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should display de items with useFetchGifs", () => {
    const gifts = [
      {
        id: "ABC",
        title: "The Title!",
        url: "https://localhost/anything.jpg",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifts,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifts.length);
  });
});
