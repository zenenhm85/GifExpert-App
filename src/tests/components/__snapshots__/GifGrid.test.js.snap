// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GifGrid should display de items with useFetchGifs 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="The Title!"
      url="https://localhost/anything.jpg"
    />
  </div>
</Fragment>
`;

exports[`Tests for GifGrid should display the component correctly 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <p>
    Loading....
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
